core/eolearn/core/eodata_io.py

```diff
@@ -248,26 +247,62 @@ def load_eopatch_content(
     temporal_selection: None | slice | list[int],
 ) -> PatchContentType:
     """A utility function used by `EOPatch.load` method."""
+    err_msg = "Feature {} is specified to be loaded but does not exist in EOPatch at " + patch_location
     file_information = get_filesystem_data_info(filesystem, patch_location, features)
-    bbox, timestamps, old_meta_info = _load_meta_features(filesystem, file_information, features, temporal_selection)
+    requested_meta = {ftype for ftype, _ in FeatureParser(features).get_feature_specifications() if ftype.is_meta()}
+
+    features_dict = _load_features(filesystem, features, temporal_selection, err_msg, file_information)
+
+    bbox = None
+    if file_information.bbox is not None:
+        bbox = FeatureIOBBox(file_information.bbox, filesystem)
+    elif FeatureType.BBOX in requested_meta and features is not Ellipsis:
+        raise IOError(err_msg.format(FeatureType.BBOX))
+
+    timestamps = None
+    if FeatureType.TIMESTAMPS in requested_meta:
+        if file_information.timestamps is not None:
+            timestamps = FeatureIOTimestamps(file_information.timestamps, filesystem, temporal_selection)
+        elif features is not Ellipsis:
+            raise IOError(err_msg.format(FeatureType.TIMESTAMPS))
+
+    return bbox, timestamps, features_dict
+
+
+def _load_features(
+    filesystem: FS,
+    features: FeaturesSpecification,
+    temporal_selection: None | slice | list[int],
+    err_msg: str,
+    file_information: FilesystemDataInfo,
+) -> dict[tuple[FeatureType, str], FeatureIO]:
+    requested_meta = {ftype for ftype, _ in FeatureParser(features).get_feature_specifications() if ftype.is_meta()}
```

Comment:
This is repeated from L254, would it make sense to pass it to the `_load_features`?

Reply:
i was thinking if maybe i should make a note there.
this one will vanish when we adjust timestamps saving (since it will no longer rely on `features`)